### PR TITLE
Integrate submission comment delete button

### PR DESCRIPTION
**Pull request by flacial**
Related to #893. Specifically, the delete feature.
 
This PR Integrates submission comment delete button into UI.

Image:

image

[Storybook](https://deploy-preview-1473--youthful-thompson-448332.netlify.app/?path=/story/components-submissioncomments--with-comments)

### CODE REVIEW

**Comment by flacial** [comment]
I'm not able to get the google verification code to login into Vercel (Changed OS). If someone may authorize it please.

File: components/SubmissionComments.tsx [hunk 1]
```diff
@@ -1,11 +1,44 @@
...
+}> = ({ comments, submission }) => {
+  const [deleteComment] = useDeleteCommentMutation()
+  const context = useContext(GlobalContext)
+  const id = context.session?.user?.id
+
+  const handleDeleteComment = (c: Comment) => {
+    const update = updateCache(
+      submission.id,
+      c.id,
+      undefined,
```
Comment (anthonykhoa) [inline components/SubmissionComments.tsx:24]: So many undefined arguments passed through. This is pretty ugly

File: components/SubmissionComments.test.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  it('Should delete comment when delete button is clicked', async () => {
+    const cache = new InMemoryCache({ addTypename: false })
+
+    cache.writeQuery({
+      query: GET_PREVIOUS_SUBMISSIONS,
+      variables: { userId: 3, challengeId: 6 },
+      data: { getPreviousSubmissions: submissionsData }
+    })
+
+    const Wrapper = ({ children }) => {
```
Comment (flacial) [inline components/SubmissionComments.test.js:77]: Could be exported into a file under `helpers/tests` as it's frequently used.

File: components/SubmissionComments.tsx (hunk 1 above)
Comment (flacial) [reply components/SubmissionComments.tsx:24]: https://github.com/garageScript/c0d3-app/pull/1474

**Review by songz — APPROVED** [review]

**Review by songz — APPROVED** [review]

**Review by anthonykhoa — APPROVED** [review]

**Review by mino323 — APPROVED** [review]

**Pull request merged**